### DISCUSSION TITLE
SqlServerDsc future and private versions?

### DISCUSSION THREAD

**Discussion by ghost** [dsccommunity/SqlServerDsc#2000] [General]
I used PowerShell 7 with Invoke-DscResource for the first time.  (I used it to configure an operator on a sql instance.)  This can save me so much time!  I don't want to use the LCM yet... perhaps never.  I was writing my own code to run from my desktop, but I can see that the SqlServerDsc module will be much easier to use.  

I do wonder about the future of this module.  Will Microsoft be replacing it with a new-fangled DSC v3 compatible module that can control sql features on servers anywhere?

I can see I will need to add some properties here and there that are not currently supported by some resources.  How should I do that?  Can I just copy the module and rename it and hack away on my private copy?  Or should I create new v2 of just the dsc resource I change and import it as part of the original module.  I have very little experience with creating modules or adding something to an existing module.  

I like DBATools, but I like the GET-TEST-SET pattern more.  If the DBATools followed the same pattern....  

PS Invoke-DscResource appears to force verbose messages.  When it loads the SqlServerDsc module, it bloats my screen with verbose messages from the load.  My only recourse is to redirect the verbose stream to $null.  Perhaps PSDesiredStateConfiguration should be hacked as well to create a non-LCM version that will survive DSC changes.

**Comment by johlju** [comment]
Thanks for using the resources. 

> I do wonder about the future of this module. Will Microsoft be replacing it with a new-fangled DSC v3 compatible module that can control sql features on servers anywhere?

That is a question for the SQL Server team, but would say the community would iterate much faster on such initiative.

**Comment by johlju** [comment]
> I can see I will need to add some properties here and there that are not currently supported by some resources. How should I do that? Can I just copy the module and rename it and hack away on my private copy? Or should I create new v2 of just the dsc resource I change and import it as part of the original module. I have very little experience with creating modules or adding something to an existing module.

Fork the repo, make changes as you see fit, and if they can contribute for the rest of the community then send in a PR. But it is always good to start to create an issue in the GitHub repo to discuss the change so it is viable.

**Comment by johlju** [comment]
> Invoke-DscResource appears to force verbose messages. When it loads the SqlServerDsc module, it bloats my screen with verbose messages from the load.

That is because how the module is written and due to an issue with how VerbosePreference is not passed between modules (by design in PowerShell). So all helper functions are currently hard-coded to output verbose messages, but there is an issue opened to adress that, and also move some stuff to debug messages instead of verbose messages.
